digitalocean/loadbalancer/resource_loadbalancer.go

```diff
@@ -432,13 +433,110 @@ func resourceDigitalOceanLoadBalancerV0() *schema.Resource {
 					},
 				},
 			},
+
 			"type": {
 				Type:         schema.TypeString,
 				Optional:     true,
 				ForceNew:     true,
 				ValidateFunc: validation.StringInSlice([]string{"REGIONAL", "GLOBAL"}, true),
 				Description:  "the type of the load balancer (GLOBAL or REGIONAL)",
 			},
+
+			"domains": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Computed:    true,
+				MinItems:    1,
+				Description: "the list of domains required to ingress traffic to global load balancer",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.NoZeroValues,
+							Description:  "domain name",
+						},
+						"is_managed": {
+							Type:        schema.TypeBool,
+							Optional:    true,
+							Default:     false,
+							Description: "flag indicating if domain is managed by DigitalOcean",
+						},
+						"certificate_id": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.NoZeroValues,
+							Description:  "certificate ID for TLS handshaking",
+						},
```

Comment:
Does this experience the same issue as with the certificates used for forwarding rules?

We had a lot of pain around Let's Encrypt certificates as their IDs change when they are renewed. This causes Terraform to prompt the user to reconcile state and attempt to reapply the old ID. So we had to migrate users from `certificate_id` to `certificate_name` being the source of truth. Check out how we approach it in `expandForwardingRules`.

https://github.com/digitalocean/terraform-provider-digitalocean/blob/007c65e4893779497d2235816a78ccc067605b84/digitalocean/loadbalancer/loadbalancer.go#L103-L113

Reply:
Managing certificates for domain would be identical to how we do the same for forwarding rules currently. cc @bbassingthwaite to correct if I'm wrong.